stream_unix.cpp

```diff
@@ -163,6 +178,11 @@ class UnixStream : public watchman_stream {
     if (cred.cr_uid == getuid() || cred.cr_uid == 0) {
       return true;
     }
+#elif defined(SO_RECVUCRED)
+    uid_t ucreduid = ucred_getruid(cred);
```

Comment:
the `unique_ptr` will mean that we use `cred.get()` here to pass the raw pointer value on.